File: Solutions/2018/Scala/src/test/scala/Day1Tests.scala
import Day1.*

class Day1Tests extends munit.FunSuite:

  test("Part1") {
    val input = Day1.parseFile("/Day1/Input.txt")
    val obtained = Part1(input)
    val expected = 508
    assertEquals(obtained, expected)
  }

  test("Part2") {
    val input = Day1.parseFile("/Day1/Input.txt")
    val obtained = Part2(input)
    val expected = 549
    assertEquals(obtained, expected)
  }

File: Solutions/2025/Java/Java/src/test/java/Day1Tests.java
import Day1.Day1;
import org.junit.jupiter.api.Test;

import static org.junit.Assert.assertEquals;

class Day1Tests {

    @Test
    void Part1TestInput() {
        var day1 = new Day1();
        var input = day1.ParseInput("Day1/TestInput.txt");
        var result = day1.Part1(input);
        assertEquals(3, result);
    }

    @Test
    void Part1() {
        var day1 = new Day1();
        var input = day1.ParseInput("Day1/Input.txt");
        var result = day1.Part1(input);
        assertEquals(969, result);
    }

    @Test
    void Part2TestInput() {
        var day1 = new Day1();
        var input = day1.ParseInput("Day1/TestInput.txt");
        var result = day1.Part2(input);
        assertEquals(6, result);
    }

    @Test
    void Part2() {
        var day1 = new Day1();
        var input = day1.ParseInput("Day1/Input.txt");
        var result = day1.Part2(input);
        assertEquals(5887, result);
    }
}